### DISCUSSION TITLE
SideSetsFromNormalsGenerator support for distributed mesh

### DISCUSSION THREAD

**Discussion by KhaledNabilSharafeldin** [idaholab/moose#21182] [Q&A General] [answered]
Hello,
When using `SideSetsFromNormalsGenerator` with `--distributed-mesh` flag, it result in an error:

```
[2] ./include/libmesh/remote_elem.h, line 123, compiled Mar 22 2022 at 20:48:06

[0] ./include/libmesh/remote_elem.h, line 123, compiled Mar 22 2022 at 20:48:06

[1] ./include/libmesh/remote_elem.h, line 123, compiled Mar 22 2022 at 20:48:06


*** ERROR ***
Error: not implemented!



*** ERROR ***
Error: not implemented!



*** ERROR ***
Error: not implemented!

--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------
[bell-a094.rcac.purdue.edu:104207] PMIX ERROR: UNREACHABLE in file server/pmix_server.c at line 2193
------------------------------------------------------------
A process or daemon was unable to complete a TCP connection
to another process:
  Local host:    bell-a094
  Remote host:   bell-a095
This is usually caused by a firewall on the remote host. Please
check that any firewall (e.g., iptables) has been disabled and
try again.

```

You can recreate this problem by running `moose/test/tests/meshgenerators/sidesets_from_normals_generator/sidesets_cylinder_normals_fixed.i` using `--distributed-mesh` flag.

Alternatives or quick solutions would be very helpful!

Thanks in advance!
Khaled

**Comment by GiudGiud** [comment, marked as answer]
If you just need to get going with adding a sideset you can load the mesh in serial, add it then, then use this new file in your distributed script.
You can also add sidesets in cubit.

**Reply by GiudGiud** [reply]
@KhaledNabilSharafeldin this mesh generator was only implemented for replicated meshes. 
In the code you can see that there is an error message for distributed meshes, but it s broken right now. I ll fix it.

**Reply by KhaledNabilSharafeldin** [reply]
I can see that, thanks, this also goes for similar functions as well, I'll look into how mesh generators create them right away/manually adding them.
Any plans to extend the functionality? or is it unlikely?

**Reply by GiudGiud** [reply]
No plans that I know of
